src/test/java/com/make/my/day/hm4/Homework04.java

```diff
@@ -139,8 +165,117 @@ public void dogMappingByAge() {
   }
 
   @Test
-  public void partitionByOddEven() {
-    int[] numbers = new int[]{2, 3, 8, 7, 8, 9, 13, 11};
+  public void partitionByEvenOdd() {
+    List<Dog> dogs = Arrays.asList(
+        new Dog("Bim", 4), new Dog("Duke", 7), new Dog("Fenrir", 120));
+
+    //TODO: make you'r realization
+    Map<Boolean, List<Dog>> result = dogs.stream().collect(Collectors.partitioningBy(dog -> dog.getAge() % 2 == 0));
+
+    Map<Boolean, List<Dog>> expected = new HashMap<>();
+    expected.put(true, Arrays.asList(new Dog("Bim", 4), new Dog("Fenrir", 120)));
+    expected.put(false, Arrays.asList(new Dog("Duke", 7)));
+
+    assertEquals(expected, result);
+  }
+
+  private enum Role{
+    ADMIN, USER, MANAGER
+  }
+
+  private class User{
+    private String email;
+    private Role role;
+
+    public User(String email, Role role) {
+      this.email = email;
+      this.role = role;
+    }
+
+    public String getEmail() {
+      return email;
+    }
+
+    public Role getRole() {
+      return role;
+    }
+  }
+
+  private class UserDTO{
+    private String email;
+
+    private List<Role> roles;
+
+    public UserDTO(String email, List<Role> roles) {
+      this.email = email;
+      this.roles = roles;
+    }
+
+    public String getEmail() {
+      return email;
+    }
+
+    public List<Role> getRoles() {
+      return roles;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      UserDTO userDTO = (UserDTO) o;
+      return Objects.equals(email, userDTO.email) &&
+          Objects.equals(roles, userDTO.roles);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(email, roles);
+    }
+
+    @Override
+    public String toString() {
+      return "UserDTO{" +
+          "email='" + email + '\'' +
+          ", roles=" + roles +
+          '}';
+    }
+  }
+
+
+  @Test
+  public void convertFromUserToUserDTOTest() {
+    List<User> usersFromDB = Arrays.asList(
+        new User("superman@epam.com", Role.ADMIN),
+        new User("superman@epam.com", Role.USER),
+        new User("superman@epam.com", Role.MANAGER),
+        new User("someone@epam.com", Role.USER),
+        new User("sonofsun@epam.com", Role.USER),
+        new User("sonofsun@epam.com", Role.MANAGER)
+    );
+
+    Map<String, List<Role>> collect = usersFromDB.stream()
```

Comment:
Make it cool, use only one stream without braking on 2 parts

Reply:
Sure! Made a proper adjustment.